### PR TITLE
Expect cupy to now support bool arrays for dlpack.

### PR DESCRIPTION
**Pull request by vyasr**
## Description

The most recent versions of cupy now support bool arrays in dlpack (see [the release notes](https://github.com/cupy/cupy/releases/tag/v11.6.0) and [the PR adding the feature](https://github.com/cupy/cupy/pull/7376)). Our DataFrame protocol tests were expecting a failure.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/tests/test_df_protocol.py
Comment (wence-) [inline python/cudf/cudf/tests/test_df_protocol.py:47]: Does this pass need to be conditioned on a cupy version?

Comment (vyasr) [reply python/cudf/cudf/tests/test_df_protocol.py:47]: Sure we can. We have not historically managed supported versions for testing consistently for any dependency aside from pandas AFAIK, but cupy is generally stable enough that maybe the few places where our tests are already conditional on cupy version are sufficient to capture all the cases we've cared about so far.

**Comment by vyasr** [comment]
> Do we need a certain version in our dependencies/conda recipes? What is the minimum version needed?
> 
> https://github.com/rapidsai/cudf/blob/45a9e825f975ca55f57f5af9523ec0b88839ff56/dependencies.yaml#L252

I don't know how well cudf will work with cupy < 11. I haven't tried that in a long time. I can try installing 9.5 and see what happens.

File: python/cudf/cudf/tests/test_df_protocol.py
Comment (bdice) [reply python/cudf/cudf/tests/test_df_protocol.py:47]: Are we able to just set a minimum cupy version in our dependencies that meets this criterion, rather than introduce conditional logic? https://github.com/rapidsai/cudf/pull/12883#issuecomment-1456397914

Comment (wence-) [reply python/cudf/cudf/tests/test_df_protocol.py:47]: Just thinking that unless (as @bdice suggests) we bump the minimum version, any environment that was previously passing will now fail these tests.

**Comment by bdice** [comment]
Great. Let’s get that bumped, we probably shouldn’t support 3 major versions of cupy, especially with cupy 12 just around the corner.

**Comment by bdice** [comment]
I didn't realize CI was blocked by this. If that is the case, I'll approve and we can merge. However, I don't think we should continue extending support to such old cupy versions. I would bump the minimum version to one that is compatible with our tests, not just the package itself.

**Review by bdice — APPROVED** [review]
Approving this to unblock CI (changes look good) but we should revisit version support for cupy in a follow-up PR so we aren't supporting 3 major versions of cupy. Supported versions should pass all tests and we should avoid taking on debt in the form of version-conditional logic.

**Review by wence- — APPROVED** [review]

**Comment by wence-** [comment]
Thanks for the digging, I agree with @bdice's suggestions.

**Comment by vyasr** [comment]
Created #12884 to continue the discussion @bdice @wence-

**Pull request merged**
